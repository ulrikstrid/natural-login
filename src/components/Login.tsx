import * as React from 'react'
import * as classNames from 'classnames'
import * as Email from '../models/Email'
import InputWithValidation from './InputWithValidation'

interface Props {}

interface State {
  username: string,
  validEmail: boolean,
  password: string
}

class Login extends React.Component<Props, State> {
  state = {
    username: '',
    password: '',
    validEmail: true
  }

  loginSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
  }

  setUsername = (inputValue: string) => {
    this.setState({ username: inputValue })
  }

  setPassword = (e: React.FormEvent<HTMLInputElement>) => {
    this.setState({ password: e.currentTarget.value })
  }

  setValidEmail = (input: string): boolean => {
    const isValid = Email.validateEmail(input)
    this.setState({ validEmail: isValid })
    return isValid
  }

  render () {
    return (
      <div className='login-container'>
        <form className='form' onSubmit={this.loginSubmit}>
          <div className='form-group'>
            <label htmlFor='username'>Username</label>
            <InputWithValidation
              type='email'
              name='username'
              value={this.state.username}
              updateValue={this.setUsername}
              validator={this.setValidEmail}
            />
            <div className={classNames({ validation: true, invalid: !this.state.validEmail })}>
              Not a valid E-mail
            </div>
          </div>
          <div className='form-group'>
            <label htmlFor='password'>Password</label>
            <input type='password' name='password' value={this.state.password} onChange={this.setPassword} />
          </div>

          <input type='submit' className='button' />
        </form>
      </div>
    )
  }
}

export default Login
