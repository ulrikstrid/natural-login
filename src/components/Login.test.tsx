import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { default as Login, setEmail, setPassword, State, SetStateCallback } from './Login'

const initalState: State = {
  email: '',
  password: '',
  validEmail: true,
  popOver: false,
  suggestedEmail: ''
}

test('renders without crashing', () => {
  const component = renderer.create(
    <Login />
  )

  const initialTree = component.toJSON()
  expect(initialTree).toMatchSnapshot()
})

test('setters should call setState', () => {
  const email = 'ulrik@example.com'
  const password = 'password'

  setEmail((cb: SetStateCallback) => {
    expect(cb(initalState)).toEqual({ email })
  }, email)

  setPassword((cb: SetStateCallback) => {
    expect(cb(initalState)).toEqual({ password })
  }, password)
})
