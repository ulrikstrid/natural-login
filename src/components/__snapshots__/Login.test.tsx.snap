exports[`test renders without crashing 1`] = `
<div
  className="login-container">
  <form
    className="form"
    noValidate={true}
    onSubmit={[Function]}>
    <div
      className="form-group">
      <label
        htmlFor="email">
        Email
      </label>
      <input
        autoComplete="off"
        name="email"
        onChange={[Function]}
        type="email"
        value="" />
      <div
        className="validation">
        Not a valid E-mail
      </div>
    </div>
    <div
      className="form-group">
      <label
        htmlFor="password">
        Password
      </label>
      <input
        autoComplete="off"
        name="password"
        onChange={[Function]}
        type="password"
        value="" />
    </div>
    <input
      className="button primary"
      formNoValidate={true}
      type="submit" />
  </form>
  <div
    className="popover">
    <p>
      You wrote 
      <b>
        
      </b>
       but we think you ment 
      <b>
        
      </b>
      .
    </p>
    <p>
      If this is correct, press \"Yes\" else press \"No\"\" and we will submit your entered email
    </p>
    <button
      className="button primary"
      onClick={[Function]}>
      Yes
    </button>
     
    <button
      className="button">
      No
    </button>
  </div>
</div>
`;
